From 1cb923fa1af3d5e7896df405f08ae6e4f89671b5 Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Tue, 30 Oct 2012 15:16:04 -0700
Subject: init: Allow for configurable init.rc path

If androidboot.initrc is passed in the kernel command line,
the path it refers to will be used for the initial init.rc
and not /init.rc.

This is to support specialized boot modes using the same
ramdisk.

Change-Id: Icfa179c422bf1d548c4ddef33c41cc89718d762d
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 init/init.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/init/init.c b/init/init.c
index f325f61..92ed88d 100755
--- a/init/init.c
+++ b/init/init.c
@@ -844,6 +844,7 @@ int main(int argc, char **argv)
     int signal_fd_init = 0;
     int keychord_fd_init = 0;
     bool is_charger = false;
+    const char *initrc_path;
 
     /* If we are called as 'modprobe' command, we run as a
      * standalone executable and reuse ueventd's logic to do the job.
@@ -922,8 +923,11 @@ int main(int argc, char **argv)
     if (!is_charger)
         property_load_boot_defaults();
 
-    INFO("reading config file\n");
-    init_parse_config_file("/init.rc");
+    initrc_path = property_get("ro.boot.initrc");
+    if (!initrc_path)
+        initrc_path = "/init.rc";
+    INFO("reading config file %s\n", initrc_path);
+    init_parse_config_file(initrc_path);
 
     action_for_each_trigger("early-init", action_add_queue_tail);
 
-- 
1.7.4.1

